src/main/java/com/_up/megastore/services/implementations/TokenService.java

```diff
@@ -0,0 +1,31 @@
+package com._up.megastore.services.implementations;
+
+import com._up.megastore.data.model.Token;
+import com._up.megastore.data.model.User;
+import com._up.megastore.data.repositories.ITokenRepository;
+import com._up.megastore.services.interfaces.ITokenService;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class TokenService implements ITokenService {
+
+    private final ITokenRepository iTokenRepository;
+
+    public TokenService(ITokenRepository iTokenRepository){
+        this.iTokenRepository = iTokenRepository;
+    }
+
+    @Override
+    public Token saveToken(User user){
+        Token token = Token.builder()
+                .user(user)
+                .build();
+        return iTokenRepository.save(token);
+    }
+
+    public UUID findUser(UUID activationToken){
+        return iTokenRepository.findById(activationToken).get().getUser().getUserId();
+    }
```

Comment:
1. Cambiémosle el nombre a algo más descriptivo
2. Está mal que hagas un findById(id).get() porque, ¿Qué pasa si el usuario nos mete un token alterado? El user asociado a ese token no se va a encontrar. Habría que manejar eso con un .orElseThrow()
3. ¿Por qué devolvemos el ID y no el user?

```suggestion
    public UUID findUserByActivationToken(UUID activationTokenId) {
        return iTokenRepository.findById(activationToken)
             .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "Token with ID " + activationTokenId + " does not exist.")
             .getUser();
    }
```

Reply:
Devolvemos el ID y no el user para que despues validemos mediante un query method, encontrar el user si y solo si el user no esta activado